client/app/pages/queries/add-to-dashboard.js

```diff
@@ -0,0 +1,72 @@
+import template from './add-to-dashboard.html';
+
+const AddToDashboardForm = {
+  controller($sce, Dashboard, currentUser, toastr, Query, Widget) {
+    'ngInject';
+
+    this.query = this.resolve.query;
+    this.vis = this.resolve.vis;
+    this.saveAddToDashbosard = this.resolve.saveAddToDashboard;
+    this.saveInProgress = false;
+
+    this.trustAsHtml = html => $sce.trustAsHtml(html);
+
+    this.onDashboardSelected = (dash) => {
+      // add widget to dashboard
+      this.saveInProgress = true;
+      this.widgetSize = 1;
+      this.selectedVis = null;
+      this.query = {};
+      this.selected_query = this.query.id;
+      this.type = 'visualization';
+      this.isVisualization = () => this.type === 'visualization';
+
+      const widget = new Widget({
+        visualization_id: this.vis && this.vis.id,
+        dashboard_id: dash.id,
+        options: {},
+        width: this.widgetSize,
+        type: this.type,
+      });
+
+      // (response)
+      widget.$save().then(() => {
+        // (dashboard)
+        this.selectedDashboard = Dashboard.get({ slug: dash.slug }, () => {});
```

Comment:
I thought it would need to be:

`Dashboard.get({slug: dash.slug}, (selectedDash) => {this.selectedDashboard = selectedDash;});`

Doesn't really matter if this works though.